### PR TITLE
Add full docs for 1.4.0

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR adds release notes for 1.4.0 and its jars.

### CODE REVIEW

File: docs/config.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ home = [ "HTML", "RSS", "SearchIndex" ]
 [menu]
   versions = [
     { name = "latest", pre = "relative", url = "../latest", weight = 1 },
+    { name = "1.4.0", pre = "relative", url = "../1.4.0", weight = 987 },
```
Comment (Fokko) [inline docs/config.toml:27]:
```suggestion
    { name = "1.4.0", pre = "relative", url = "../1.4.0", weight = 988 },
```

File: landing-page/content/common/multi-engine-support.md [hunk 2]
```diff
@@ -66,10 +66,11 @@ Each engine version undergoes the following lifecycle stages:
...
-| 3.2        | Maintained         | 0.13.0                  | {{% icebergVersion %}} | [iceberg-spark-runtime-3.2_2.12](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-spark-runtime-3.2_2.12/{{% icebergVersion %}}/iceberg-spark-runtime-3.2_2.12-{{% icebergVersion %}}.jar) |
+| 3.1        | End of Life        | 0.12.0                  | 1.3.1                  | [iceberg-spark-runtime-3.1_2.12](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-spark-runtime-3.1_2.12/1.3.1/iceberg-spark-runtime-3.1_2.12-1.3.1.jar) [1]                               |
+| 3.2        | Deprecated         | 0.13.0                  | {{% icebergVersion %}} | [iceberg-spark-runtime-3.2_2.12](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-spark-runtime-3.2_2.12/{{% icebergVersion %}}/iceberg-spark-runtime-3.2_2.12-{{% icebergVersion %}}.jar) |
```
Comment (Fokko) [inline landing-page/content/common/multi-engine-support.md:70]: Nice! Let's remove it from the repository soon, such a waste of CI.

**Comment by aokolnychyi** [comment]
@amogh-jahagirdar @Fokko @nastra @rdblue @danielcweeks @RussellSpitzer @flyrain

File: landing-page/content/common/release-notes.md [hunk 3]
```diff
@@ -26,10 +26,10 @@ disableSidebar: true
 The latest version of Iceberg is [{{% icebergVersion %}}](https://github.com/apache/iceberg/releases/tag/apache-iceberg-{{% icebergVersion %}}).
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:27]: Should we also add the new bundles to the list?

```
* [{{% icebergVersion %}} aws-bundle Jar](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-aws-bundle/{{% icebergVersion %}}/iceberg-aws-bundle-{{% icebergVersion %}}.jar)
* [{{% icebergVersion %}} gcp-bundle Jar](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-gcp-bundle/{{% icebergVersion %}}/iceberg-gcp-bundle-{{% icebergVersion %}}.jar)
* [{{% icebergVersion %}} azure-bundle Jar](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-azure-bundle/{{% icebergVersion %}}/iceberg-azure-bundle-{{% icebergVersion %}}.jar)
```
(tested the paths)

**Review by Fokko — APPROVED** [review]
Should we add the bundles too? I like them a lot! Apart from that it looks good! 👍

File: landing-page/content/common/release-notes.md (hunk 3 above)
Comment (aokolnychyi) [reply landing-page/content/common/release-notes.md:27]: Sounds good, added below.

File: docs/config.toml (hunk 1 above)
Comment (aokolnychyi) [reply docs/config.toml]: Good catch.

**Pull request merged**